ksql-engine/src/main/java/io/confluent/ksql/codegen/SqlToJavaVisitor.java

```diff
@@ -500,11 +500,31 @@ private String visitBooleanComparisonExpression(final ComparisonExpression.Type
       final Pair<String, Schema> value = process(node.getValue(), context);
       switch (node.getSign()) {
         case MINUS:
-          // this is to avoid turning a sequence of "-" into a comment (i.e., "-- comment")
-          final String separator = value.getLeft().startsWith("-") ? " " : "";
-          return new Pair<>("-" + separator + value.getLeft(), value.getRight());
+          if (DecimalUtil.isDecimal(value.getRight())) {
```

Comment:
This method is getting larger with this enhancement. Consider refactoring. 

You could either refactor each case into a method, e.g.

```java
switch (node.getSign()) {
case Minus:  return visitArithmeticMinus(...);
case Plus: return visitArithmeticPlus(...);
}
```

Or pull out the decimal as a special case, e.g.

```java
if (DecimalUtil.isDecimal(value.getRight())) {
   return visitDecimalArithmeticUnary(..);
}

// old code here
}
```

Personally, I prefer the first.